### PR TITLE
DOC: fix DataFrame.sample doctests and reformat the docstring

### PR DESCRIPTION
**Pull request by Moisan**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Based on #22459. Fix the docstring for DataFrame.sample. I also updated `ci/doctests.sh`.

I was on the fence if I should remove mentions of `Panels` since it is deprecated.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -4340,7 +4340,7 @@ def sample(self, n=None, frac=None, replace=False, weights=None,
             being sampled.
             If weights do not sum to 1, they will be normalized to sum to 1.
             Missing values in the weights column will be treated as zero.
-            inf and -inf values not allowed.
+            `inf` and `-inf` values not allowed.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: If this is code literals, then it should be double-backticks and quoted

    ``'inf'``

which is a bit noise. How about "infinite values are not allowed.?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -4350,58 +4350,57 @@ def sample(self, n=None, frac=None, replace=False, weights=None,
 
         Returns
         -------
-        A new object of same type as caller.
+        Series or DataFrame:
+            A new object of same type as caller.
+
+        See Also
+        --------
+        numpy.choice: Generates a random sample from a given 1-D numpy array.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: `numpy.random.choice`?

**Review by TomAugspurger — APPROVED** [review]

File: pandas/core/generic.py [hunk 3]
```diff
@@ -4350,58 +4350,57 @@ def sample(self, n=None, frac=None, replace=False, weights=None,
 
         Returns
         -------
-        A new object of same type as caller.
+        Series or DataFrame:
```
Comment (datapythonista) [inline pandas/core/generic.py]: The colon at the end is not needed in this case.

File: pandas/core/generic.py [hunk 4]
```diff
@@ -4350,58 +4350,57 @@ def sample(self, n=None, frac=None, replace=False, weights=None,
...
+            A new object of same type as caller.
```
Comment (datapythonista) [inline pandas/core/generic.py]: It'd be nice to briefly describe what is returned, not only the type.

File: pandas/core/generic.py [hunk 5]
```diff
@@ -4350,58 +4350,57 @@ def sample(self, n=None, frac=None, replace=False, weights=None,
...
-        0   -0.038497
-        1    1.820773
-        2   -0.972766
-        3   -1.598270
-        4   -1.095526
-        dtype: float64
-        >>> df = pd.DataFrame(np.random.randn(50, 4), columns=list('ABCD'))
+        >>> df = pd.DataFrame({'A': range(0, 500, 10),
+        ...                    'B': range(0, 100, 2),
+        ...                    'C': range(0, 250, 5)})
```
Comment (datapythonista) [inline pandas/core/generic.py]: I don't think we need so much data, and being arbitrary. Something like the animal datasets we've been using with 6 or 8 rows should be better to illustrate the resample method.

**Review by datapythonista — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
@datapythonista

**Review by datapythonista — APPROVED** [review]
Looks great, and very good example.

**Comment by datapythonista** [comment]
Thanks for another excellent PR @Moisan

**Pull request merged**